Unit test fails with GSL2

<!-- TEMPLATE FOR BUG REPORTS -->
### Expected behavior

All unit tests and system test pass.
### Actual behavior

GSL 2.2 breaks yet another unit test. The results below are from OS X.

```
$ ctest -R CurveFittingTest_DiffRotDiscreteCircleTest  --verbose
UpdateCTestConfiguration  from :/Users/svh/Documents/MantidProject/build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/Users/svh/Documents/MantidProject/build/DartConfiguration.tcl
Test project /Users/svh/Documents/MantidProject/build
Constructing a list of tests
Done constructing a list of tests
Checking test dependency graph...
Checking test dependency graph end
test 1154
    Start 1154: CurveFittingTest_DiffRotDiscreteCircleTest

1154: Test command: /usr/local/Cellar/cmake/3.6.1/bin/cmake "-E" "chdir" "/Users/svh/Documents/MantidProject/build/bin/Testing" "/Users/svh/Documents/MantidProject/build/bin/CurveFittingTest" "DiffRotDiscreteCircleTest"
1154: Test timeout computed to be: 300
1154: FrameworkManager-[Notice] Welcome to Mantid 3.7.20160902.2041
1154: FrameworkManager-[Notice] Please cite: http://dx.doi.org/10.1016/j.nima.2014.07.029 and this release: http://dx.doi.org/10.5286/Software/Mantid
1154: Running 9 tests.Fit-[Notice] Fit started
1154: Fit-[Notice] Fit successful, Duration 0.01 seconds
1154: .Fit-[Notice] Fit started
1154: Fit-[Notice] Fit successful, Duration 0.01 seconds
1154: .Fit-[Notice] Fit started
1154: Fit-[Notice] Fit successful, Duration 0.02 seconds
1154: .Fit-[Notice] Fit started
1154: Fit-[Notice] Fit successful, Duration 0.02 seconds
1154: .Fit-[Notice] Fit started
1154: Fit-[Notice] Fit successful, Duration 0.00 seconds
1154: ...Fit-[Notice] Fit started
1154: Fit-[Notice] Fit successful, Duration 0.63 seconds
1154: 
1154: In DiffRotDiscreteCircleTest::testDiffRotDiscreteCircle:
1154: /Users/svh/Documents/MantidProject/mantid/Framework/CurveFitting/test/Functions/DiffRotDiscreteCircleTest.h:298: Error: Expected (fitalg_structure_factor->getParameter("Radius") == 1.567) up to 1.567 * 0.05 (0.0783), found (1.7477 != 1.5670)
1154: /Users/svh/Documents/MantidProject/mantid/Framework/CurveFitting/test/Functions/DiffRotDiscreteCircleTest.h:300: Error: Expected (fitalg_structure_factor->getParameter("Decay") == 7.567) up to 7.567 * 0.05 (0.3783), found (9.1182 != 7.5670)
1154: Failed 1 of 9 tests
1154: Success rate: 88%
1/1 Test #1154: CurveFittingTest_DiffRotDiscreteCircleTest ...***Failed    1.07 sec

0% tests passed, 1 tests failed out of 1

Total Test time (real) =   1.17 sec

The following tests FAILED:
    1154 - CurveFittingTest_DiffRotDiscreteCircleTest (Failed)
Errors while running CTest
```
### Steps to reproduce the behavior

Build Mantid on OS X and run `ctest -R CurveFittingTest_DiffRotDiscreteCircleTest  --verbose`.
### Platforms affected

OS X, possibly Linux with Clang

--- a/Framework/CurveFitting/test/Functions/DiffRotDiscreteCircleTest.h
+++ b/Framework/CurveFitting/test/Functions/DiffRotDiscreteCircleTest.h
@@ -239,7 +239,7 @@ public:
         "(composite=Convolution,FixResolution=true,NumDeriv=true;name=Gaussian,"
         "Height=1,PeakCentre=0,Sigma=20,ties=(Height=1,PeakCentre=0,Sigma=20);("
         "name=DiffRotDiscreteCircle,N=3,NumDeriv=true,Q=0.5,Intensity=47.014,"
-        "Radius=1.567,Decay=7.567))";
+        "Radius=1.567,Decay=0.07567))";
 
     // Initialize the fit function in the Fit algorithm
     Algorithms::Fit fitalg;
@@ -258,7 +258,7 @@ public:
                      "name=Gaussian,Height=1,PeakCentre=0,Sigma=20,ties=("
                      "Height=1,PeakCentre=0,Sigma=20);(name="
                      "DiffRotDiscreteCircle,N=3,NumDeriv=true,Q=0.5,Intensity="
-                     "10.0,Radius=1.567,Decay=20.0))";
+                     "20.0,Radius=1.567,Decay=0.1))";
     fitalg.setProperty("Function", funtion_string);
     fitalg.setProperty("InputWorkspace", data_workspace);
     fitalg.setPropertyValue("WorkspaceIndex", "0");
@@ -296,10 +296,10 @@ public:
                     47.014 * 0.05); // allow for a small percent variation
     TS_ASSERT_DELTA(fitalg_structure_factor->getParameter("Radius"), 1.567,
                     1.567 * 0.05); // allow for a small percent variation
-    TS_ASSERT_DELTA(fitalg_structure_factor->getParameter("Decay"), 7.567,
+    TS_ASSERT_DELTA(fitalg_structure_factor->getParameter("Decay"), 0.07567,
                     7.567 * 0.05); // allow for a small percent variation
     // std::cout << "\nGOAL: Intensity = 47.014,  Radius = 1.567,  Decay =
-    // 7.567\n"; // only for debugging purposes
+    // 0.07567\n"; // only for debugging purposes
     // std::cout << "OPTIMIZED: Intensity = " <<
     // fitalg_structure_factor->getParameter("Intensity") << "  Radius = " <<
     // fitalg_structure_factor->getParameter("Radius") << "  Decay = " <<

